demos/machine_translation_demo/python/machine_translation_demo.py

```diff
@@ -72,25 +72,24 @@ class TranslationEngine:
     def __init__(self, model_xml, model_bin, device, output_name):
         log.info('OpenVINO Inference Engine')
         log.info('\tbuild: {}'.format(get_version()))
-        ie = IECore()
+        core = Core()
 
         log.info('Reading model {}'.format(model_xml))
-        self.net = ie.read_network(
-            model=model_xml,
-            weights=model_bin
-        )
-        self.net_exec = ie.load_network(self.net, device)
+        self.model = core.read_model(model_xml, model_bin)
+        compiled_model = core.compile_model(self.model, args.device)
+        self.infer_request = compiled_model.create_infer_request()
         log.info('The model {} is loaded to {}'.format(model_xml, device))
-        self.output_name = output_name
-        assert self.output_name != "", "there is not output in model"
+        self.input_tensor_name = "tokens"
```

Comment:
do we need also check input tensor name?

Reply:
It will be unified as described in https://github.com/openvinotoolkit/open_model_zoo/pull/3036#discussion_r786097341. For the current version this check is not required. It's the intel model and there is should not happen some issue with missing input name